plugins/juju/04known_bugs

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+import logging
+import subprocess
+
+from common import helpers
+from common.constants import JUJU_LOG_PATH
+
+log = logging.getLogger("juju.04known-bugs")
+
+
+def detect_bug1910958():
+    """Unit fails to start complaining there are members in the relation."""
+    result = {"bug": "LP: #1910958",
+              "description": ("Unit fails to start complaining there are "
+                              "members in the relation."),
+              }
+    pattern = (
+        'manifold worker returned unexpected error: failed to initialize '
+        'uniter for "[A-Za-z0-9\\-]+": cannot create relation state tracker: '
+        'cannot remove persisted state, relation \\d+ has members'
+    )
+
+    cmd = ["grep", "-r", "-P", "-H", "--include=*.log", "--files-with-match",
+           pattern, JUJU_LOG_PATH]
+    try:
+        output = subprocess.check_output(cmd,
+                                         universal_newlines=True)
+        log.debug("cmd: %s, exit code: 0", cmd)
+        result["files_with_matches"] = output.strip().split("\n")
+    except subprocess.CalledProcessError as ex:
+        log.debug("cmd: %s, exit code: %s", cmd, ex.returncode)
+        if ex.returncode == "2":  # file not found
+            log.warn(str(ex))
+        elif ex.returncode == "1":  # pattern not found
+            pass
+
+        return None
+
+    return result
+
+
+KNOWN_ISSUES = [detect_bug1910958]
+
+
+if __name__ == "__main__":
+
+    issues_found = []
```

Comment:
e.g. https://pastebin.ubuntu.com/p/Ffws7BWr7D/

Reply:
done. This is how it will look like in the revision of the patch:

```
./hotsos.sh  --juju tests/unit/fake_data_root/
...
juju:
  machines: {}
  charm-versions: []
  units:
    stopped:
      - rabbitmq-server-2
      - rabbitmq-server-3
    lxd:
      - ceph-osd-no-fixed-wal-7
      - nova-compute-0
  launchpad:
    - 1910958:
        description: Unit fails to start complaining there are members in the relation.
        match: 'tests/unit/fake_data_root/var/log/juju/unit-rabbitmq-server-3.log:2021-02-17
          08:18:44 ERROR juju.worker.dependency engine.go:671 "uniter" manifold worker
          returned unexpected error: failed to initializ...
  
          tests/unit/fake_data_root/var/log/juju/unit-rabbitmq-server-3.log:2021-02-17
          08:20:34 ERROR juju.worker.dependency engine.go:671 "uniter" manifold worker
          returned unexpected error: failed to initializ...
  
          tests/unit/fake_data_root/var/log/juju/unit-rabbitmq-server-2.log:2021-02-17
          08:18:44 ERROR juju.worker.dependency engine.go:671 "uniter" manifold worker
          returned unexpected error: failed to initializ...
  
          tests/unit/fake_data_root/var/log/juju/unit-rabbitmq-server-2.log:2021-02-17
          08:20:34 ERROR juju.worker.dependency engine.go:671 "uniter" manifold worker
          returned unexpected error: failed to initializ...'

INFO: see --help for more display options
```